Sources/PromotedFirebaseRemoteConfig/ClientConfig+PromotedFirebaseRemoteConfig.swift

```diff
@@ -0,0 +1,207 @@
+import CryptoKit
+import Firebase
+import Foundation
+
+#if SWIFT_PACKAGE
+import FirebaseRemoteConfig
+import PromotedCore
+#endif
+
+extension ClientConfig {
+
+  func merge(
+    from remoteConfig: RemoteConfig,
+    messages: inout PendingLogMessages
+  ) {
+    var dictionary: [String: String] = [:]
+    for key in remoteConfig.allKeys(from: .remote) {
+      dictionary[key] = remoteConfig.configValue(forKey: key).stringValue
+    }
+    merge(from: dictionary, messages: &messages)
+  }
+
+  func merge(
+    from dictionary: [String: String],
+    messages: inout PendingLogMessages
+  ) {
+    var remainingKeys = Set(dictionary.keys)
+    let mirror = Mirror(reflecting: self)
+
+    // It's much safer to iterate over all keys in ClientConfig,
+    // even though the dictionary should only contain a subset of
+    // these keys.
+    // If we pass a non-existent key to NSObject.setValue(), it
+    // throws an ObjC exception that we can't catch in Swift.
+    // This could cause a runtime crash.
+    // Iterating through ClientConfig's keys ensures that the keys
+    // we pass to setValue are valid.
+    for child in mirror.children {
+      let optionalChildLabel = child.label
+      // `assertInValidation` is a local debugging flag.
+      if optionalChildLabel == "assertInValidation" { continue }
+      guard let childLabel = optionalChildLabel else {
+        messages.warning(
+          "Child with no label: \(String(describing: child))",
+          visibility: .public
+        )
+        continue
+      }
+
+      let key = "ai_promoted_" + childLabel.toSnakeCase()
```

Comment:
Interesting.  Why is this transformation needed?  Do you map the key directly to other constants in the code?

Reply:
Drew and I wanted to make the keys match between iOS and Android as best we could. So we decided that snake case would work best to standardize on between the two.

We wanted the keys to be namespaced in case clients want to use their own Firebase accounts to control remote config.

The keys map directly to the property names in the object.